src/app/(auth)/signup/page.tsx

```diff
@@ -23,27 +25,33 @@ export default function SignUp() {
 
   const validEmail = (e: string) => e !== '' && isEmail(e);
 
-  const handleSignUp = async () => {
+  const handleSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
     if (!auth) return;
-    setEmailError(validEmail(email) ? '' : 'Invalid Email');
-    setPasswordError(password !== '' ? '' : 'Invalid Password');
+    setEmailError(validEmail(email) ? '' : 'Invalid email.');
+    setPasswordError(password !== '' ? '' : 'Invalid password.');
     if (!validEmail(email) || password === '') {
       setErrorMessage('');
       return;
     }
     if (!passwordComplexity) {
-      setPasswordError('Password must meet complexity requirements');
+      setPasswordError('Password must meet complexity requirements.');
+      return;
+    }
+    if (password !== confirmPassword) {
+      setErrorMessage('Passwords do not match.');
       return;
     }
     setEmailError('');
     setPasswordError('');
-    const { error } = await auth.signUp(email, password, {
+    const error = await auth.signUp(email, password, {
       emailRedirectTo:
         'https://immigration-justice-project.vercel.app/email-verified',
     });
 
     if (error) {
       setErrorMessage(error.message);
```

Comment:
Since we want to return an `AuthResponse` again, let's revert back to the previous code:
`const { error } = ...`